database/migrations/default/1729616317798_set_more_spatial_attributes/up.sql

```diff
@@ -0,0 +1,134 @@
+        --
+        -- get engineering area
+        --
+        new.engineering_area_id = (
+            select
+                engineering_area_id
+            from
+                geo.engineering_areas
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'engineering_area_id: % compared to previous: %', new.engineering_area_id, old.engineering_area_id;
+        --
+        -- get signal engineer area
+        --
+        new.signal_engineer_area_id = (
+            select
+                signal_engineer_area_id
+            from
+                geo.signal_engineer_areas
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'signal_engineer_area_id: % compared to previous: %', new.signal_engineer_area_id, old.signal_engineer_area_id;
+        --
+        -- get zip code
+        --
+        new.zipcode = (
+            select
+                zipcode
+            from
+                geo.zip_codes
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'zipcode: % compared to previous: %', new.zipcode, old.zipcode;
+        --
+        -- get apd sector
+        --
+        new.apd_sector_id = (
+            select
+                primary_key
+            from
+                geo.apd_sectors
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'apd_sector_id: % compared to previous: %', new.apd_sector_id, old.apd_sector_id;
+        --
+        -- check if is_non_coa_roadway
+        --
+        if (new.in_austin_full_purpose or new.rpt_city_id = 22) then
+            new.is_non_coa_roadway = st_contains((select geometry from geo.non_coa_roadways), new.position);
+            raise debug 'is_non_coa_roadway: % compared to previous: %', new.is_non_coa_roadway, old.is_non_coa_roadway;
+        end if;
+        else
+            raise debug 'reseting spatial attributes due to null latitude and/or longitude values';
+            -- nullify position column
+            new.position = null;
+            -- reset location id
+            new.location_id = null;
+            -- use city id to determine full purpose jurisdiction
+            new.in_austin_full_purpose = coalesce(new.rpt_city_id = 22, false);
+            raise debug 'setting in_austin_full_purpose based on city id: %', new.in_austin_full_purpose;
+            -- reset council district
+            new.council_district = null;
+            -- reset engineering area
+            new.engineering_area_id = null;
+            -- reset signal eng area
+            new.signal_engineer_area_id = null;
+            -- reset zip code
+            new.zipcode = null;
+            -- reset apd_sector
+            new.apd_sector_id = null;
+            -- reset is_non_coa_roadway
+            new.is_non_coa_roadway = true;
```

Comment:
yes. a crash is not on a COA roadway until proven otherwise.

i've gone back and forth on whether or not we should call this field `is_coa_roadway` instead, and default it to `false`. Xavier convinced me to go with `is_non_coa_roadway`, but coming back to this i think the double negative  is too confusing and we should go with `is_coa_roadway` instead.